### ISSUE TITLE
Lower CIDEr values on CIDEr Optimization 

### ISSUE THREAD

**Issue by goelarushi** [ruotianluo/self-critical.pytorch#136]
I am training my captioning model for 32 epochs (with CE loss) and then doing SCST training. But my rewards increase in the negative and my CIDEr decreases. Any possible reasons for this? 
Thanks for the help in advance. 

![reward](https://user-images.githubusercontent.com/35184420/66914147-0ffda400-f00e-11e9-855a-a3875ed62ed8.png)

**Comment by zhengyi123456** [comment]
Could you provide the details of your scores, like CIDEr or Bleu and the model you used.

**Comment by ruotianluo** [comment]
Are you using the same hyperparameters as in the readme? Can you check your learning rate?

**Comment by kangkang59812** [comment]
Did you solve it? thanks

**Comment by goelarushi** [comment]
Hi everyone. Thanks for your reply. 
@ruotianluo - I am using it for a little bit different setting but if I make the learning rate very small (5e-6), the cider starts to increase for a few epochs but then starts decreasing again. Do you suggest a much smaller learning rate? My cider before optimization is 114.5 and increases only upto 115.1.

**Comment by ruotianluo** [comment]
@goelarushi can you tell me what is your setting? what feature, what model are you using?

**Comment by goelarushi** [comment]
@ruotianluo, I am using the bottom-up features and a model similar to the original BUTD paper.

**Comment by ruotianluo** [comment]
@goelarushi if you run the script in README.md, is that doing the right thing?

**Comment by goelarushi** [comment]
Yes, when I try to run the same setting as yours it works. But when I try to take a part of your cider optimization to my code and setting, it behaves differently. my loss and net reward oscillate between -0.75 and -0.96 respectively. Thanks for your help in advance.

**Comment by ruotianluo** [comment]
It's hard for me to guess what is wrong if you are using your own code. Are you running on COCO?

**Comment by goelarushi** [comment]
I understand. I will try to debug and I hope i get close to the solution soon. I am running on COCO Karpathy splits.

**Comment by fawazsammani** [comment]
@goelarushi are you using the code [here](https://github.com/sgrvinod/a-PyTorch-Tutorial-to-Image-Captioning)?

**Comment by goelarushi** [comment]
Hi @fawazsammani . Thanks for your reply. I am indeed using similar code. Do let me know if you encounter while using CIDEr optim. I see my reward doesn't go down much after a some epochs. 
Thanks.
